pom.xml

```diff
@@ -135,6 +135,22 @@
     </dependency>
   </dependencies>
 
+  <build>
+    <plugins>
+      <plugin>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <version>3.0.0-M3</version>
```

Comment:
I'm going to commit this in a separate PR, so I'll just have you rebase against that if you don't mind.

Reply:
Sure. The tests don't work on my Mac anyway. Says something is wrong with the `log.dir` property.

Do you get the same? I wasn't able to quickly find any similar errors searching the spring kafka issues

Reply:
@cricket007 When building from master, I don't get those issues. Have you rebased this commit against the master changes?

Reply:
I cloned your branch locally, and I do get this error when running a build:

```
14:04:48.441 [main] ERROR kafka.server.BrokerMetadataCheckpoint - Failed to read meta.properties file under dir /var/folders/b7/j31znfln6gvc4sghhcq4nwqr0000gn/T/kafka-1473154000058593224/meta.properties due to /var/folders/b7/j31znfln6gvc4sghhcq4nwqr0000gn/T/kafka-1473154000058593224/meta.properties
14:04:48.443 [main] ERROR kafka.server.KafkaServer - Fail to read meta.properties under log directory /var/folders/b7/j31znfln6gvc4sghhcq4nwqr0000gn/T/kafka-1473154000058593224
java.nio.file.NoSuchFileException: /var/folders/b7/j31znfln6gvc4sghhcq4nwqr0000gn/T/kafka-1473154000058593224/meta.properties
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
	at java.nio.file.Files.newByteChannel(Files.java:361)
	at java.nio.file.Files.newByteChannel(Files.java:407)
	at java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:384)
	at java.nio.file.Files.newInputStream(Files.java:152)
	at org.apache.kafka.common.utils.Utils.loadProps(Utils.java:560)
	at kafka.server.BrokerMetadataCheckpoint.liftedTree2$1(BrokerMetadataCheckpoint.scala:63)
	at kafka.server.BrokerMetadataCheckpoint.read(BrokerMetadataCheckpoint.scala:62)
	at kafka.server.KafkaServer$$anonfun$getBrokerIdAndOfflineDirs$1.apply(KafkaServer.scala:668)
	at kafka.server.KafkaServer$$anonfun$getBrokerIdAndOfflineDirs$1.apply(KafkaServer.scala:666)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)
	at kafka.server.KafkaServer.getBrokerIdAndOfflineDirs(KafkaServer.scala:666)
	at kafka.server.KafkaServer.startup(KafkaServer.scala:209)
	at kafka.utils.TestUtils$.createServer(TestUtils.scala:132)
	at kafka.utils.TestUtils.createServer(TestUtils.scala)
	at org.springframework.kafka.test.EmbeddedKafkaBroker.afterPropertiesSet(EmbeddedKafkaBroker.java:223)
	at org.springframework.kafka.test.rule.EmbeddedKafkaRule.before(EmbeddedKafkaRule.java:109)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

Perhaps something added when configuring the admin client as well as just producers/consumers is misconfigured in the `EmbeddedKafka`?